engine/function/src/templates/Numeric.ftl

```diff
@@ -3378,6 +3390,600 @@ public class Numeric {
+        }
+
+        // skip over nulls
+        ${pt.primitive} next = Math.nextAfter(start, direction);
+        return isNull(next) ? Math.nextAfter(next, direction) : next;
+     }
+     </#if>
+
+     <#if pt.valueType.isFloat>
+     /**
+      * Returns the floating-point number adjacent to the argument in the direction of positive infinity.
+      *
+      * @param x the value.
+      * @return the floating-point number adjacent to the argument in the direction of positive infinity.
+      *    If the value is null, returns null.
+      */
+     static public ${pt.primitive} nextUp(${pt.primitive} x) {
+        if(isNull(x)) {
+            return ${pt.null};
+        }
+
+        // skip over nulls
+        ${pt.primitive} next = Math.nextUp(x);
+        return isNull(next) ? Math.nextUp(next) : next;
+     }
+     </#if>
+
+     <#if pt.valueType.isFloat>
+     /**
+      * Returns the floating-point number adjacent to the argument in the direction of negative infinity.
+      *
+      * @param x the value.
+      * @return the floating-point number adjacent to the argument in the direction of negative infinity.
+      *    If the value is null, returns null.
+      */
+     static public ${pt.primitive} nextDown(${pt.primitive} x) {
+        if(isNull(x)) {
+            return ${pt.null};
+        }
+
+        // skip over nulls
+        ${pt.primitive} next = Math.nextDown(x);
+        return isNull(next) ? Math.nextDown(next) : next;
+     }
+     </#if>
+
+    /**
+     * Converts an angle measured in radians to an approximately equivalent angle measured in degrees.
+     *
+     * @param x the angle in radians
+     * @return the measurement of the angle x in degrees.  If the value is null, returns null.
+     */
+    static public double toDegrees(${pt.primitive} x) {
+        if(isNull(x)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.toDegrees(x);
+    }
+
+    /**
+     * Converts an angle measured in degrees to an approximately equivalent angle measured in radians.
+     *
+     * @param x the angle in degrees
+     * @return the measurement of the angle x in radians.  If the value is null, returns null.
+     */
+    static public double toRadians(${pt.primitive} x) {
+        if(isNull(x)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.toRadians(x);
+    }
+
+    <#if pt.valueType.isInteger>
+    /**
+     * Returns the value of the argument as an int, throwing an exception if the value overflows an int.
+     *
+     * @param x the value.
+     * @return the value as an int.  If either value is null, returns null.
+     * @throws ArithmeticException if the result overflows.
+     */
+    static public int toIntExact(${pt.primitive} x) {
+        if (isNull(x)) {
+            return NULL_INT;
+        }
+
+        <#if pt.primitive == "byte" || pt.primitive == "short" || pt.primitive == "int" >
+        return x;
+        <#else>
+        return Math.toIntExact(x);
```

Comment:
This is a weird case but bear with me.
I don't like what happens when pt.primitive is `long` and the long value happens to be `Integer.MIN_VALUE`

This is a perfectly valid long value and in particular is not Deephaven NULL.

However, in this case this method would return Integer.MIN_VALUE, i.e. a deephaven NULL.

So in summary it took a long value that was *not* a Deephaven NULL and turned it into an integer value that *is* a Deephaven NULL.

I think I would test the result of Math.toIntExact with IsNull and throw an ArtihmeticException if it returns true.  That is, this case should be treated like an overflow.